src/ui/Api.js

```diff
@@ -27,11 +27,33 @@ export default class Api {
 
     if (typeof id !== "string" || typeof count !== "number") {
       throw new Error(
-        "dealV2 requires a deck id and a number representing how many cards should be dealt",
+        "dealV2 requires a deck id and a number representing how many cards should be dealt"
       );
     }
-    const resp = await fetch(`/api/v2/deck/${id}/deal`, { method: "POST" });
-    const response = await resp.json();
-    return response;
+    // const resp = await fetch(`/api/v2/deck/${id}/deal`, { method: "POST" });
+    const cards = await Promise.all(
```

Comment:
You definitely don't need to do multiple promises for v2. You should be able to just do something like this:

```
let newCards = await fetch(`/api/v2/deck/${id}/deal?count={numCardsToDeal}`);
```

In other words, specify the number of cards you need in the request, and then let the server return that many cards in a single request as a list.